aten/src/ATen/native/Onehot.cpp

```diff
@@ -0,0 +1,35 @@
+#include <ATen/ATen.h>
+
+namespace at { namespace native {
+
+Tensor one_hot(const Tensor &self, int64_t num_classes) {
+    auto shape = self.sizes().vec();
```

Comment:
Can you check that the type of self is integral and throw an exception if it is not? This makes the error reporting a little nicer

Reply:
@zou3519 What is the best way to check if it is integral? Currently it looks ugly...

Reply:
!is_floating_point(self) && !is_complex(self) should work, but it is not the best...

Reply:
Oh, you could call this macro: https://github.com/pytorch/pytorch/blob/7a61306031648fc4e62fe9e5868893ebf9ae6fb6/aten/src/ATen/Dispatch.h#L54-L66 and have it do nothing; it would check to see if the tensor is integral.